File: LeetCode/Google Questions/Interview Process/Java/Unique Email Addresses.java
class Solution {
    public int numUniqueEmails(String[] emails) {
        if (emails == null) {
            return 0;
        }

        Set<String> addresses = new HashSet<>();
	  
        for (String email : emails) {
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < email.length(); i++) {
                char ch = email.charAt(i);
                switch (ch) {
                    case '.':
                        break;
                    case '+':
                        while (ch != '@') {
                            i++;
                            ch = email.charAt(i);
                        }
                        sb.append(email.substring(i));
                        i = email.length();
                    break;
                    case '@':
                        sb.append(email.substring(i));
                        i = email.length();
                        break;
                    default:
                        sb.append(ch);
                }
            }
            addresses.add(sb.toString());
        }
        return addresses.size();
    }
} 

File: LeetCode/Google Questions/Interview Process/Unique Email Addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        # count = 0
        locals = []
        domains = []
        
        for email in emails:
            for index, char in enumerate(email):
                if char == "@":
                    locals.append(email[0:index])
                    domains.append(email[index:])
                    continue
        
        valids = set()
        
        for index, localname in enumerate(locals):
            # localname = local
            localname = localname.split("+")[0]
            localname = localname.replace(".","")
            localname = (localname + "@" +  domains[index])
            valids.add(localname)
        
        return len(valids)